fdbserver/DDTxnProcessor.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * DDTxnProcessor.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FOUNDATIONDB_DDTXNPROCESSOR_H
+#define FOUNDATIONDB_DDTXNPROCESSOR_H
+
+#include "fdbserver/Knobs.h"
+#include "fdbserver/DataDistribution.actor.h"
+
+class IDDTxnProcessor {
+public:
+	// get the source server and complete source server for range
+	virtual Future<std::tuple<std::vector<UID>, std::vector<UID>>> getSourceServersForRange(
```

Comment:
Nit: It looks like the type can be `std::pair` instead of `std::tuple` here. IMO it may be clearer if the return type conveyed information about what is being returned (e.g. `IDDTxnProcessor::SourceServers`).

Reply:
Here it has to be a tuple because storeTuple returns a tuple. I want to make `storeTuple` return tuple for its convenience for storing arbitrary number of variables like `storeTuple(future, a, b, c...)`.